### PR TITLE
Add BertLargeTF to the new benchmark suite

### PR DESCRIPTION
**Pull request by pzread**
Add BertLargeTF to the new benchmark suite. (added to legacy suite in #12032)

After #12032, I think it's time to add a full support for configurable ImportedModel for custom flags (which originally I tried to keep it simple). This change introduces `ImportConfig`, which describes the dialect and import flags to import a model.

benchmarks: x86_64, cuda

### CODE REVIEW

File: build_tools/python/e2e_test_framework/definitions/iree_definitions.py [hunk 1]
```diff
@@ -98,55 +98,90 @@ class ModuleExecutionConfig(object):
...
-  # Name of an IREE supported input type (--iree-input-type).
-  dialect_name: str
-
-
-# Please use and update the next id when adding a new dialect, so we won't reuse
-# the old deprecated id.
-# Next ID: 4
-class MLIRDialectType(_MLIRDialectPair, Enum):
+# Value should be the name of an IREE supported input type (--iree-input-type).
+class MLIRDialectType(Enum):
```
Comment (GMNGeoffrey) [inline build_tools/python/e2e_test_framework/definitions/iree_definitions.py:109]: Why was this change to the enum reverted?

File: build_tools/python/e2e_test_framework/definitions/iree_definitions.py [hunk 2]
```diff
@@ -98,55 +98,90 @@ class ModuleExecutionConfig(object):
...
+
+
+# Placeholder to be replaced with entry function name when outputting the actual
+# flag list.
+IMPORT_CONFIG_ENTRY_FUNCTION_PLACEHOLDER = "$ENTRY_FUNCTION_PLACEHOLDER"
+
+
+@serialization.serializable(type_key="iree_import_configs")
+@dataclass(frozen=True)
+class ImportConfig(object):
```
Comment (GMNGeoffrey) [inline build_tools/python/e2e_test_framework/definitions/iree_definitions.py:123]: It's more than just flags, right? Don't we need to say what tool is being used?

File: build_tools/python/e2e_test_framework/definitions/iree_definitions.py (hunk 1 above)
Comment (pzread) [reply build_tools/python/e2e_test_framework/definitions/iree_definitions.py:109]: MLIRDialectType can now be part of the ImportConfig, so no need to have a dedicated id for it.

Basically if we are going to output a different dialect, we just create a new ImportConfig for it.

File: build_tools/python/e2e_test_framework/definitions/iree_definitions.py (hunk 2 above)
Comment (pzread) [reply build_tools/python/e2e_test_framework/definitions/iree_definitions.py:123]: It was decided when generating rules previously. I think it's a good idea to move it here. Done.

**Review by GMNGeoffrey — APPROVED** [review]

**Pull request merged**